connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -151,4 +160,19 @@ public void run() {
         }
     }
 
+    public boolean isPaused() {
+        return this.targetState.get() == TargetState.PAUSED;
+    }
+
+    public void transitionTo(TargetState state) {
+        TargetState oldState = this.targetState.getAndSet(state);
+        if (state != oldState && running.get()) {
+            // FIXME: this is racy
```

Comment:
I'm confused by this comment and why `this.targetState` is an `AtomicReference`. Isn't this only ever going to be called by one thread (from the Herder processing target state changes)?